### ISSUE TITLE
Implement rate limiting for processing The Trexler File

### ISSUE THREAD

**Issue by holdenhinkle** [department-of-veterans-affairs/va.gov-team#76624]
## Background
Over 20k jobs almost instantly fail when first processed in sidekiq because the Lighthouse API endpoint we're hitting errors due to its rate limit. All 20k jobs will never be processed because retries are set to 25 by default (I'm guessing we're using the default?). I need to come up with a better way to handle this.

## Tasks
- [ ] Implement rate-limiting using the rate-limit feature in Sidekiq Enterprise 

## Acceptance Criteria
- [ ] _What will be created or happen as a result of this?_

---
#### How to configure this issue
1. Label with practice area (`backend`, `frontend`,`ux`, `design`, `research`)
2. Label for MVP work? (`mvp`)
3. Add an estimate 
4. Attach an Epic (what body of work is this a part of?)

**Comment by oddball-lindsay** [comment]
This is required to refresh addresses after adding in VSOfficers to Find a Rep 2.0

**Comment by holdenhinkle** [comment]
When you use Sidekiq Enterprise's rate limiting feature in combination with batches, it's important to understand how both mechanisms interact, especially in scenarios where a large number of jobs are batched and rate limited. Here's an overview of how the process works and what it means for your job execution times:

### Batching Jobs in Sidekiq
- **Batching**: When you create a batch in Sidekiq, you're grouping a set of jobs that will be executed as part of that batch. Sidekiq tracks the progress of all jobs within the batch and can execute callbacks upon completion of all jobs in the batch.
- **Batch Execution**: Jobs within a batch are enqueued and executed according to their individual rules and the available worker concurrency. Batching itself does not impose a delay on job enqueuing; it's about grouping jobs for tracking and callback purposes.

### Applying Rate Limiting
- **Rate Limiting**: When you apply rate limiting to a job class using Sidekiq Enterprise, you're specifying limits on how often jobs of that class can be executed. If a job is enqueued that would exceed the rate limit, its execution is delayed until the rate limit permits it.
- **Job Execution Delay**: The rate limiting mechanism doesn't prevent jobs from being enqueued; it controls their execution. If a job is subject to rate limiting, it will be enqueued immediately but will wait in the queue until it's allowed to be processed according to the defined rate limits.

### Interaction Between Batching and Rate Limiting
- **Batch Processing Time**: When you batch a large number of jobs that are also subject to rate limiting, the initial enqueuing of jobs into the batch happens quickly. However, the overall processing time of the batch will be extended because each job's execution is governed by the rate limit. The batch is considered complete when all jobs have been executed, which can take a considerable amount of time if the jobs are rate limited.
- **Throttling Effect**: The rate limiting effectively throttles the processing of batched jobs, spreading their execution over a longer period. This can help manage load on your resources or comply with external API rate limits but also means that the completion of the entire batch will be delayed.

### Practical Implications
- **Job Creation vs. Execution**: The job that creates batches of jobs will complete its execution once it has enqueued all the jobs into batches. The time it takes to create and enqueue these jobs is generally short. However, the total time until all jobs in the batches are processed will be longer, depending on the rate limits applied.
- **System Load and Throughput Management**: By using rate limiting, you can manage the load on your system more effectively. This approach allows you to spread out job execution over time, preventing spikes in load that could affect system performance or stability.

In summary, the job that creates batches of jobs will complete the enqueuing process relatively quickly. The rate limiting primarily affects the execution time of the jobs within the batches, extending the total time it takes for the entire batch to be processed. This mechanism provides a way to manage system load and ensure that job execution is spread out to meet specific rate limits or resource utilization goals.

**Comment by oddball-lindsay** [comment]
With this new approach, estimating 12 hours to process

**Comment by holdenhinkle** [comment]
Sidekiq rate-limit documentation - https://github.com/sidekiq/sidekiq/wiki/Ent-Rate-Limiting

**Comment by holdenhinkle** [comment]
PR - https://github.com/department-of-veterans-affairs/vets-api/pull/15605

**Comment by holdenhinkle** [comment]
It's working perfectly - better than expected:

It's already processed > 500 jobs in ~20 mins. This would have taken MUCH longer (days) before (to get through all the agents data). And it's working through the 3 batches (agents, attorneys, representatives) one at a time.

<img width="2000" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMHA3QXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--cef4fd65213dd65b9fba95be805bfd88521bf035/image.png" alt="image.png" />
